CHANGELOG.md

```diff
@@ -0,0 +1,122 @@
+<!-- markdownlint-disable MD024 -->
+
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+### Added
+
+- Pull request template with comprehensive contribution guidelines
+- MIGRATION.md for tracking breaking changes and upgrade paths
+- CHANGELOG.md for tracking all notable changes
+
+### Changed
+
+### Deprecated
+
+### Removed
+
+### Fixed
+
+### Security
+
+---
+
+## Example Format
+
+Below is an example of how to document changes. Remove this section once you
+have real entries.
+
+## [1.2.0] - 2025-01-15
+
+### Added
+
+- New `/v2/learner` API endpoint with improved data model
+- Support for user roles in the authentication system
+- Database migration script for version 1.2.0
+- Configuration validation on startup
+
+### Changed
+
+- **BREAKING:** Environment variable naming convention now uses `LIF_DATABASE_*`
+  prefix instead of `LIF_DB_*`
+- Improved error messages in GraphQL API responses
+- Updated Python dependencies to latest compatible versions
+
+### Deprecated
+
+- `/v1/learner` endpoint will be removed in version 2.0.0 (use `/v2/learner` instead)
+
+### Removed
+
+- **BREAKING:** Removed deprecated `/v1/learner` endpoint (deprecated in 1.0.0)
```

Comment:
I get it's just an example, but the removed note is in contradiction to the dreprecation note .